joomla/joomla-cms
# [4.0] Migrate data for media field in repeatable field

Pull Request for Issue # .

### Summary of Changes
When data for a **Repeatable** field from Joomla 3 migrated to a **Subfields** on Joomla 4, the value for media field type is stored in String, not in JSON format as expected, thus it throws some warnings when the field is being displayed (see https://github.com/joomla/joomla-cms/pull/32611#issuecomment-816985442 )

This PR attempts to solve that.

### Testing Instructions
1. Install Joomla 3.10.0. You can download the nightly build package here https://developer.joomla.org/nightlies/Joomla_3.10.0-alpha6-dev-Development-Full_Package.zip to install it.
2. Access to **Content -> Fields**, create a **Repeatable** field. This field should contains at least one **Media** field for testing purpose (see screenshot for example)
![repeatable_field](https://user-images.githubusercontent.com/977664/114392961-adb23f00-9bc3-11eb-89ab-260e91acf780.png)
3. Create one or two articles. When you add article, navigate to **Fields** tab, enter data for the repeatable field which you created above. See screenshot for example data
![screenshot](https://user-images.githubusercontent.com/977664/114393186-f10cad80-9bc3-11eb-9c72-0ad525f2a152.png)

4. Upgrade your site to latest Joomla 4.0-dev package (generate for this PR). Just download the updated package here https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/33111/downloads/42298/Joomla_4.0.0-beta8-dev+pr.33111-Development-Update_Package.zip, then go to **Components** -> **Joomla ! Update**, look **Upload & Update** tab and install that update package.
5. Check the articles on upgraded site, edit these article, check and make sure data in Fields tab is there.
![field_data](https://user-images.githubusercontent.com/977664/114393678-7beda800-9bc4-11eb-8dcd-f123f3a7263f.png)

6. Set the article to **Featured** so that it is displayed on homepage of the site
7. Access to the site in the frontend, see the article with custom fields data displayed. Make sure there are no warning. Unfortunately, the image which you select in the field for article could not be displayed because of a bug (is solved in a different PR), so ignore it. Just make sure there are no warnings and the test could be marked as success

![Home](https://user-images.githubusercontent.com/977664/114393745-945dc280-9bc4-11eb-8e62-fc4de5f9fa1c.png)
